### ISSUE TITLE
"Segmentation fault" error when running the flash_ilo4 tool on Ubuntu Live.

### ISSUE THREAD

**Issue by marcandrer** [kendallgoto/ilo4_unlock#21]
I've tried to patch 2 servers (HP DL160 G8 & DL360 G9). Both servers gave me the same "Segmentation fault" error when running the "sudo ./flash_ilo4 --direct" command. I tried running it from Ubuntu Live CD 21.10 and the latest Ubuntu Live CD version.

I'm not sure if this could affect anything, but I ran the build.sh scripts from WSL - Windowos Subsystem for Linux (Ubuntu distro).
I got the BIN file properly patched, as far as I can tell.

Is there any other way to install the patched BIN?

Thank you!

**Comment by kendallgoto** [comment]
hi @marcandrer - were you able to solve this? if so, can you share more details in case it comes up for someone else? I haven't heard of this issue before.

**Comment by marcandrer** [comment]
@kendallgoto, I'm not sure what my issue was, but downloading the already patched 2.77 firmware from a reddit post, then downloading CP027911.scexe, and manually extracting the files with "sh ./CP027911.scexe --unpack=ilo_250" fixed the issue.

I don't remember getting any errors when I ran the "build.sh" scripts in WSL, so I'm not sure why that wasn't working.

Porting everything to Python 3 might really help as I found it is getting really hard to install anything Python 2 related.
